sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1010,21 +1018,44 @@ class AstBuilder extends DataTypeAstBuilder
       // [EMPTY]
       query
     } else {
-      throw QueryParsingErrors.combinationQueryResultClausesUnsupportedError(ctx)
+      val allClauses = "ORDER BY/SORT BY/DISTRIBUTE BY/CLUSTER BY"
+      if (forPipeOperators) {
+        throw QueryParsingErrors.clausesWithPipeOperatorsUnsupportedError(ctx, allClauses)
```

Comment:
We should always call `combinationQueryResultClausesUnsupportedError` here, as this is not related to pipe. We don't support this combination in both pipe and classic SQL.

Reply:
Sounds good, this is done.